geospaas_harvesting/harvesters.py

```diff
@@ -83,24 +89,60 @@ def harvest(self):
                 break
 
 
-class PODAACHarvester(Harvester):
-    """Harvester class for PODAAC data (NASA)"""
+class WebDirectoryHarvester(Harvester):
+    """
+    class for harvesting online data sources that rely on webpages (and most of the time on opendap)
+    """
+    ingester = None
+    crawler = None
+
     def _create_crawlers(self):
-        return [
-            crawlers.OpenDAPCrawler(url, time_range=(self.get_time_range()))
-            for url in self.config['urls']
-        ]
+        try:
+            isinstance(self.crawler, crawlers.Crawler)
```

Comment:
You again mix two things `isinstance` and `self.crawler` inside try/exept. This is incorrect. As I write previously your try/exept whould be very explicit and contain only one thing prone to errors. In particlular `isinstance()` does not raise TypeError. It returns True or False. I guess `self.crawler` may raise exception when `crawler` is not part of `self`, right? But then you should use:
```python
if not hasattr(self, 'crawler'):
    raise HarvesterConfigurationError(...)
```

Reply:
Actually, `hasattr(self, 'crawler')` is always true since `crawler` is a class attribute, so there is no need to test it.
You can just do this:
```python
if not isinstance(self.crawler, crawlers.Crawler):
    raise HarvesterConfigurationError(...)
```
(`isinstance()` will return False if `self.crawler` is None, which will be the case if it is not overloaded in the child class)

Reply:
Those lines of code are written for these kind of tests:
![image](https://user-images.githubusercontent.com/62345618/90397331-feef4a80-e097-11ea-81a8-994e1ef2577f.png)

Although is isinstance return True or False but the usage of it will raise the TypeError for the first and the second above tests. Is it OK with this explanation and these tests? I have changed it to isinstance based on @aperrin66 recommendation.

Reply:
@akorosov please mention above and give me your feedback

Reply:
@aperrin66 
I have already to do in that way: https://github.com/nansencenter/django-geo-spaas-harvesting/pull/20#discussion_r471452890

But the previous tests that instance from this harvester is nagging and they wont pass if we write it like that!

Reply:
@aperrin66 
Do you want me to change the previous tests based on it?

Reply:
@aperrin66 @akorosov 
But I still does not cover those lines with these three new tests:
![image](https://user-images.githubusercontent.com/62345618/90398140-3a3e4900-e099-11ea-95fb-9fa9e0641a61.png)


Then I am wrong!

Reply:
Hum, I tried it locally and the tests pass with the use of `isinstance()` I described above

Reply:
But I have tried them once again, and the same result just like before:
![image](https://user-images.githubusercontent.com/62345618/90398926-6dcda300-e09a-11ea-96ec-9ee5fbab6a93.png)

Reply:
Oh my bad, `issubclass()` should be used, not `isinstance()`, since self.crawler and self.ingester contain classes

Reply:
The general problems are:
1. The tests are not realistic. Why would a crawler suddenly become 222?
2. In the code unrealistic scenario is tested. Why would `self.crawler` become not a `crawlers.Crawler` instance?

Normally `self.crawler` can be either None or `crawlers.Crawler` instance right?
Then we should check only if it is not None.
```python
in class definition:

if self.crawler is None:
    raise HarvesterConfigurationError("The class of crawler has not been specified properly") 

in tests:
        wdh = harvesters.WebDirectoryHarvester()
        with self.assertRaises(HarvesterConfigurationError):
            wdh._create_crawlers()

other behaviour will raise other types of errors. But we cannot check them all.
```